examples/step-61/step-61.cc

```diff
@@ -0,0 +1,885 @@
+                                {
+                                   const Tensor<1,dim> phi_k_u = fe_face_values_rt[velocities].value (k, q);
+                                   velocity_face += beta(k)*phi_k_u;
+                                 }
+                             exact_velocity_face = exact_velocity.value (fe_face_values_rt.quadrature_point (q));
+                             difference_velocity_face_sqr = (velocity_face*normal - exact_velocity_face*normal)*
+                        		                            (velocity_face*normal - exact_velocity_face*normal);
+                              L2_err_flux_face_sqr_local += difference_velocity_face_sqr*fe_face_values_rt.JxW(q);
+                             }
+                      err_flux_each_face = L2_err_flux_face_sqr_local/(face_length)*(cell_area);
+                      err_flux_face += err_flux_each_face;
+                     }
+                  L2_err_flux_sqr +=  err_flux_face;
+      }
+
+  // After adding up errors over all cells, we take square root and get the $L_2$ errors of velocity and flux.
+  const double L2_err_velocity_cell = sqrt(L2_err_velocity_cell_sqr_global);
+  std::cout<< "L2_error_vel "<<L2_err_velocity_cell <<std::endl;
+  const double L2_err_flux_face = sqrt(L2_err_flux_sqr);
+  std::cout<< "L2_error_flux "<<L2_err_flux_face <<std::endl;
+
+}
+
+
+ // @sect4{WGDarcyEquation::output_results}
+
+ // We have 2 sets of results to output:  the interior solution
+ // and the skeleton solution. We use <code>DataOut</code> to graph interior results.
+ // The graphical output for the skeleton results is done by using
+ // the <code>DataOutFaces</code> class.
+ template <int dim>
+ void WGDarcyEquation<dim>::output_results () const
+ {
+  DataOut<dim> data_out;
+  data_out.attach_dof_handler (dof_handler);
+  data_out.add_data_vector (solution, "Pressure_Interior");
+  data_out.build_patches (fe.degree);
+  std::ofstream output ("Pressure_Interior.vtk");
+  data_out.write_vtk (output);
+
+  DataOutFaces<dim> data_out_face(false);
+  std::vector<DataComponentInterpretation::DataComponentInterpretation>
+  face_component_type(2, DataComponentInterpretation::component_is_scalar);
+  data_out_face.add_data_vector (dof_handler,
+                                 solution,
+                                 "Pressure_Edge",
+                                 face_component_type);
+  data_out_face.build_patches (fe.degree);
+  std::ofstream face_output ("Pressure_Edge.vtk");
+  data_out_face.write_vtk (face_output);
+ }
+
+
+ // @sect4{WGDarcyEquation::run}
+
+ // This is the final function of the main class. It calls the other functions of our class.
+ template <int dim>
+ void WGDarcyEquation<dim>::run ()
+ {
+   std::cout << "Solving problem in " << dim << " space dimensions." << std::endl;
+   make_grid();
+   setup_system ();
+   assemble_system ();
+   solve ();
+   process_solution ();
+   postprocess();
```

Comment:
It is a bit confusing that `process_solutions` calculates errors and the reconstruction takes place in `postprocess`.
It would be nice to compute a globally reconstructed solution (using the RaviartThomas space for the sake of simplicity) and then calculate errors in a separate method called `compute_errors` or the like.

Reply:
The problem here is that the RT space that we reconstruct in is a broken space, i.e., it is defined on each cell separately without the need to be continuous across edges. But we don't have a finite element description of this broken space in deal.II, so I can't generate a global reconstruction, store it in a vector, and use it later on.

This is #6059.